server/application/application.go

```diff
@@ -1091,6 +1091,7 @@ func (s *Server) validateAndNormalizeApp(ctx context.Context, app *appv1.Applica
 	if app.GetName() == "" {
 		return fmt.Errorf("resource name may not be empty")
 	}
+	app.Name = argo.TruncateLabelName(app.Name)
```

Comment:
As discussed, can you check if this truncation can happen only for the label and not name.

Reply:
> I think they expect us to trim the label and not the app name. Need to clarify.

I tested by specifying a label name and app name separately.
`argocd app create guestbook12345678901abcdkirankumarlol -l label=guestbook-12345678901-abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz1234567890987654321234567890lol --repo https://github.com/argoproj/argocd-example-apps.git --path guestbook --dest-server https://kubernetes.default.svc --dest-namespace default`

It did not consider the label name. It added the app name as a label name. Based on this, I presume the current implementation is right.